xsct.c

```diff
@@ -208,7 +209,7 @@ int main(int argc, char **argv)
     }
     else
     {
-        if (temp < 0)
+        if (temp == -1)
```

Comment:
```suggestion
        if (temp == TEMPERATURE_INIT)
```

should `#define TEMPERATURE_INIT (-1)` above and then change line 77 to set `int temp = TEMPERATURE_INIT;`.